pkg/apis/policy/types.go

```diff
@@ -228,6 +228,13 @@ type AllowedHostPath struct {
 	// `/foo` would allow `/foo`, `/foo/` and `/foo/bar`
 	// `/foo` would not allow `/food` or `/etc/foo`
 	PathPrefix string
+
+	// when set to true, will allow host volumes matching the pathPrefix if all volume mounts are readOnly.
+	// If the container specifically requests to run with a non-read only volume that matches the path prefix
+	// then the PSP should deny the pod.
```

Comment:
"when set to true, will allow host volumes matching the pathPrefix only if all volume mounts are readOnly."

drop "If the container specifically requests to run with a non-read only volume that matches the path prefix then the PSP should deny the pod."

along with "If set to false the container may run with a read-only root volume if it wishes but it will not be forced to.", we should **very** explicitly warn that writeable hostPath directory volumes allow containers to write to the filesystem in ways that let them traverse the host filesystem outside the pathPrefix.

the only real way I could see allowedHostPaths being useful with readOnly=false if they also allowed restricting to a particular type (e.g. "you may mount /foo/bar, but only if the type is Socket):

https://github.com/kubernetes/kubernetes/blob/65a819338a640faf6ddee9d6eea4559a20106c56/pkg/apis/core/types.go#L515-L522

https://github.com/kubernetes/kubernetes/blob/65a819338a640faf6ddee9d6eea4559a20106c56/pkg/apis/core/types.go#L505-L512

that could be something to shoot for in 1.12